client/state/themes/test/actions.js

```diff
@@ -482,4 +489,168 @@ describe( 'actions', () => {
 			} );
 		} );
 	} );
+
+	describe( '#pollThemeTransferStatus', () => {
+		const siteId = '2211667';
+
+		useNock( ( nock ) => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.get( `/rest/v1.1/sites/${ siteId }/automated-transfers/status/1` )
+				.reply( 200, { status: 'complete', message: 'all done', themeId: 'mood' } )
+				.get( `/rest/v1.1/sites/${ siteId }/automated-transfers/status/2` ).thrice()
+				.reply( 200, { status: 'stuck', message: 'jammed' } )
+				.get( `/rest/v1.1/sites/${ siteId }/automated-transfers/status/3` ).twice()
+				.reply( 200, { status: 'progress', message: 'in progress' } )
+				.get( `/rest/v1.1/sites/${ siteId }/automated-transfers/status/3` )
+				.reply( 200, { status: 'complete', message: 'all done', themeId: 'mood' } )
+				.get( `/rest/v1.1/sites/${ siteId }/automated-transfers/status/4` )
+				.reply( 400, 'something wrong' );
+		} );
+
+		it( 'should dispatch success on status complete', () => {
+			pollThemeTransferStatus( siteId, 1 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: THEME_TRANSFER_STATUS_SUCCESS,
+					siteId,
+					transferId: 1,
+					status: 'complete',
+					message: 'all done',
+					themeId: 'mood',
+				} );
+			} );
+		} );
+
+		it( 'should time-out if status never complete', ( done ) => {
+			pollThemeTransferStatus( siteId, 2, 10, 25 )( spy ).then( () => {
+				done();
+				expect( spy ).to.have.been.calledWith( {
+					type: THEME_TRANSFER_STATUS_FAILURE,
+					siteId,
+					transferId: 2,
+					error: 'client timeout',
+				} );
+			} );
+		} );
+
+		it( 'should dispatch status update', ( done ) => {
+			pollThemeTransferStatus( siteId, 3, 20 )( spy ).then( () => {
+				done();
+				// Two 'progress' then a 'complete'
+				expect( spy ).to.have.been.calledThrice;
+				expect( spy ).to.have.been.calledWith( {
+					type: THEME_TRANSFER_STATUS_SUCCESS,
+					siteId: siteId,
+					transferId: 3,
+					status: 'progress',
+					message: 'in progress',
+					themeId: undefined,
+				} );
+				expect( spy ).to.have.been.calledWith( {
+					type: THEME_TRANSFER_STATUS_SUCCESS,
+					siteId: siteId,
+					transferId: 3,
+					status: 'complete',
+					message: 'all done',
+					themeId: 'mood',
+				} );
+			} );
+		} );
+
+		it( 'should dispatch failure on receipt of error', () => {
+			pollThemeTransferStatus( siteId, 4 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWithMatch( {
+					type: THEME_TRANSFER_STATUS_FAILURE,
+					siteId,
+					transferId: 4,
+				} );
+				expect( spy ).to.have.been.calledWith( sinon.match.has( 'error', sinon.match.truthy ) );
+			} );
+		} );
+	} );
+
+	describe( '#initiateThemeTransfer', () => {
```

Comment:
> I ran into odd `network access not allowed` problems when trying to organize that properly

Hmm, that means `nock` has been called the allowed number of times with a request already and then defers to the underlying network (or would, if it was allowed). Maybe `persist` would help here?